asserts/snap_asserts.go

```diff
@@ -99,9 +99,101 @@ func buildSnapBuild(assert assertionBase) (Assertion, error) {
 	}, nil
 }
 
+// SnapRevision holds a snap-revision assertion, asserting the properties of a
+// snap release by a store authority.
+type SnapRevision struct {
+	assertionBase
+	sequence  uint64
+	timestamp time.Time
+}
+
+// SnapID returns the snap id of the built snap.
+func (assert *SnapRevision) SnapID() string {
+	return assert.Header("snap-id")
+}
+
+// SnapDigest returns the digest of the built snap.
+func (assert *SnapRevision) SnapDigest() string {
+	return assert.Header("snap-digest")
+}
+
+// Sequence returns the sequence number.
+func (assert *SnapRevision) Sequence() uint64 {
+	return assert.sequence
+}
+
+// SnapBuild returns the digest of the associated snap-build.
+func (assert *SnapRevision) SnapBuild() string {
+	return assert.Header("snap-build")
+}
+
+// DeveloperID returns the developer's ID.
+func (assert *SnapRevision) DeveloperID() string {
+	return assert.Header("developer-id")
+}
+
+// Timestamp returns the revision timestamp.
+func (assert *SnapRevision) Timestamp() time.Time {
+	return assert.timestamp
+}
+
+// Implement further consistency checks.
+func (assert *SnapRevision) checkConsistency(db *Database, acck *AccountKey) error {
+	// TODO: check the associated snap-build exists.
+	// TODO: check the associated snap-build's digest.
+	// TODO: check developer's account-key exists.
+	if !acck.isKeyValidAt(assert.timestamp) {
+		return fmt.Errorf("snap-revision timestamp outside of signing key validity")
+	}
+	return nil
+}
+
+func buildSnapRevision(assert assertionBase) (Assertion, error) {
+	_, err := checkMandatory(assert.headers, "snap-id")
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: more parsing/checking of this here?
+	_, err = checkMandatory(assert.headers, "snap-digest")
+	if err != nil {
+		return nil, err
+	}
+
+	sequence, err := checkUint(assert.headers, "sequence", 64)
```

Comment:
@niemeyer yes, I think even the system design doc has it as "snap-revision" now, though I think we have a couple other assertions there that have a sequence still that need renaming too